proposals/init.md

```diff
@@ -0,0 +1,490 @@
+likely makes sense as a general feature. There is no reason it should be 
+limited to just `init` members.
+
+**Resolution**
+There will be no warning on consumption of `init` fields and properties.
+
+LDM wants to have a broader discussion on the idea of required fields and
+properties. That may cause us to come back and reconsider our position on
+`init` members and validation.
+
+### Allow init as a type modifier
+In the same way the `readonly` modifier can be applied to a `struct` to 
+automatically declare all fields as `readonly`, the `init` only modifier can
+be declared on a `struct` or `class` to automatically mark all fields as `init`.
+This means the following two type declarations are equivalent:
+
+```cs
+struct Point
+{
+    public init int X;
+    public init int Y;
+}
+
+// vs. 
+
+init struct Point
+{
+    public int X;
+    public int Y;
+}
+```
+
+**Resolution** 
+This feature is too *cute* here and conflicts with the `readonly struct` 
+feature on which it is based. The `readonly struct` feature is simple in that 
+it applies `readonly` to all members: fields, methods, etc ... The 
+`init struct` feature would only apply to fields. This actually ends up making
+it confusing for users. 
+
+Given that `init` is only valid on certain aspects of a type we rejected the 
+idea of having it as a type modifier.
+
+## Allow init as a field modifier
+In the same way `init` can server as a property accessor it could also serve as
+a designation on fields to give them similar behaviors as `init` properties.
+That would allow for the field to be assigned before construction was complete
+by the type, derived types or object initializers.
+
+```cs
+class Student
+{
+    public init string FirstName;
+    public init string LastName;
+}
+
+var s = new Student()
+{
+    FirstName = "Jarde",
+    LastName = "Parsons",
+}
+
+s.FirstName = "Jared"; // Error FirstName is readonly
+```
+
+In metadata these fields would be marked in the same way as `readonly` fields 
+but with an additional attribute or modreq to indicate they are `init` style
+fields. 
+
+**Resolution**
+LDM agrees this proposal is sound but overal the scenario felt disjoint from 
+properties. The decision was to proceed with properties only for now but this 
+will be reconsidered if there is significant customer feedback that justifies
+the scenario.
+
+## Considerations
+
+### Compatibility
+The `init` feature is designed to be compatible with existing data types. 
+Specifically it is meant to be a completely additive change for data which is
+`readonly` today but desires more flexible object creation semantics.
+
+For example consider the following type:
+
+```cs
+class Name
+{
+    public readonly string First;
+    public readonly string Last;
+
+    public Name(string first, string last)
+    {
+        First = first;
+        Last = last;
+    }
+}
+```
+
+It is not a breaking change to use `init` in place of `readonly` here:
```

Comment:
True for structs as well?

Reply:
This entire section is too field centric. Need to pretty much completely overhaul this.